File: web-admin/src/types/employee.ts
export interface Employee {
  employee_id: number;
  department_id?: number;
  employee_code: string;
  employee_name: string;
  email?: string;
  phone?: string;
  position?: string;
  status: 'Active' | 'Inactive' | 'Resigned';
  department_name?: string;
}

export interface EmployeeCreate {
  employee_code: string;
  employee_name: string;
  department_id?: number;
  email?: string;
  phone?: string;
  position?: string;
  status?: 'Active' | 'Inactive' | 'Resigned';
}

export interface Department {
  department_id: number;
  department_name: string;
  description?: string;
  status: string;
}

File: app/models/employee.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base

class Employee(Base):
    __tablename__ = "employees"

    employee_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    department_id = Column(Integer, ForeignKey("departments.department_id"), nullable=True)
    employee_code = Column(String(30), unique=True, index=True, nullable=False)
    employee_name = Column(String(100), nullable=False)
    email = Column(String(100), nullable=True)
    phone = Column(String(20), nullable=True)
    position = Column(String(100), nullable=True)
    status = Column(String(20), default="Active", nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    # Relationships
    department = relationship("Department", back_populates="employees")
    users = relationship("User", back_populates="employee")
    assignments = relationship("AssetAssignment", back_populates="employee")
